### PR TITLE
[OPPRO-243] Support spark33 shim layer (pr1)

### PR DESCRIPTION
**Pull request by jinchengchenghh**
This PR will change much, so separate it to many PRs to avoid continuous conflict

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -36,6 +36,7 @@
 
   <properties>
     <spark32.version>3.2.2</spark32.version>
+    <spark33.version>3.3.0</spark33.version>
```
Comment (zzcclp) [inline pom.xml:39]: Now Spark 3.3.1 is release voting, we can wait for it ?

Comment (jinchengchenghh) [reply pom.xml:39]: I compare the two versions about shimlayer, it is very similiar, after spark3.3.1 release, I will upgrade from 3.3.0 to it

**Comment by zhztheplayer** [comment]
@jinchengchenghh Can we consider binding Gluten minor versions to different Spark minor versions?

`gluten-1.0.x` rely on `spark-3.2.x`
`gluten-1.1.x` rely on `spark-3.3.x`

,etc.

**Comment by zhztheplayer** [comment]
I am thinking about if we can either support different Spark version by different Gluten version, e.g.

> gluten-1.0.x -> spark-3.2.x
> gluten-1.1.x -> spark-3.3.x

or we just support as many Spark versions in a single gluten bundled build, by programmatically loading different shim codes at runtime after checking Spark version running in current JVM.

Have you happened to make some investigations on such topics?

**Comment by jinchengchenghh** [comment]
And delta lake newest version only support newest spark version
<html>
<body>


Delta Lake version | Apache Spark version
-- | --
2.1.x | 3.3.x
2.0.x | 3.2.x
1.2.x | 3.2.x
1.1.x | 3.2.x
1.0.x | 3.1.x
0.7.x and 0.8.x | 3.0.x
Below 0.7.0 | 2.4.2 - 2.4.<latest>


</body>
</html>

**Comment by jinchengchenghh** [comment]
We will support the latest 2 spark version, other older version can use our older version, so if spark release 3.4.x, we will not support 3.2 and upgrade gluten version to a new version

**Comment by zhztheplayer** [comment]
> Do you think use gluten-spark3.2_2.12-1.0.0-snapshot-jar-with-dependencies.jar is a good idea?

That sounds reasonable to me.

> We will support the latest 2 Spark version

Yes, by all means marking the supported Spark version somewhere if we decide to only support a specific version in a specific jar.

**Comment by zhztheplayer** [comment]
> And delta lake newest version only support newest spark version

Also noted that they do bind their minor version to Spark minor version. So they can roll their minor versions in parallel.

**Pull request closed without merging**